### ISSUE TITLE
User with multiple devices

### ISSUE THREAD

**Issue by icemancast** [Edujugon/PushNotification#119]
Using the notification channel how would i handle if a user has multiple devices? Currently I have a users table and a devices table that's a one to many. The notification setup shows:

/**
 * Route notifications for the Apn channel.
 *
 * @return string|array
 */
public function routeNotificationForApn()
{
    return $this->ios_push_token;
}

Which I assume would go in the users model. But I don't have a token for users I have a devices table that holds a users device. Any one user could have an iphone, ipad or any other device. Currently I pull all devices and send that list for a push which is working but i can't do stuff for instance of counting the users unread messages to badge as the list is all devices in the database. Does that make sense? Should i be iterating over the users and then sending their devices only? that doesn't seem like I should but just wondering.

**Comment by ExpDev07** [comment]
The setup should be pretty easy once we know that the Apn Notification Channel (granted you're using https://github.com/laravel-notification-channels/apn) can accept an array of recipients. Just make sure your ``Device`` model has the ``ios_push_token`` attribute, and then you can simply define your method as such:

```php
public function routeNotificationForApn()
{
    // Simply return an array of the push tokens (multiple recipients).
    return $this->devices->map(function ($device) {
        $device->ios_push_token;
    });
}
```

**Comment by icemancast** [comment]
Okay, yeah it does have a token it's called something different but believe that will still work. This goes in the user model correct?

**Comment by ExpDev07** [comment]
@icemancast Yes. The value that you return is the recipient of the notification, or if you return an array, recipients.

**Comment by icemancast** [comment]
How would you then get the feedback request if using this so I can unregister devices?

**Comment by ExpDev07** [comment]
Just delete the device row from the database?

**Comment by icemancast** [comment]
yes the feedback() method on a push gives back the failed tokens so I soft delete them from the database. I guess what I am trying to get to is mass notify however I can't send the badge(1) per user when i mass send the tokens through a push. 

I don't think i should iterate through each user and then grab their devices to send the unread badge number to.

**Comment by ExpDev07** [comment]
You can iterate through all users and send a notification, but for this you should use a job service (e.g redis).

**Comment by icemancast** [comment]
Okay @ExpDev07 and there should not be a rate limit for apple or android when doing this? I already have notifications working except for this part of the badge. I tried to do through laravel notification channel but didn't work that way and will likely refactor later.
